src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Extensions/ProjectExtensions.cs

```diff
@@ -67,10 +67,10 @@ public static async Task<VersionStamp> GetVersionAsync(this Project project, Can
                 AnalyzerConfigDocument? bestAnalyzerConfigDocument = null;
                 foreach (var analyzerConfigDocument in project.AnalyzerConfigDocuments)
                 {
-                    var analyzerConfigDirectory = PathUtilities.GetDirectoryName(analyzerConfigDocument.FilePath);
-                    // Suppression should be removed or addressed https://github.com/dotnet/roslyn/issues/41636
+                    // Analyzer config documents always have full paths, so GetDirectoryName will not return null.
```

Comment:
Isn't it legal for an AnalyzerConfigDocument to have a non-absolute path? I think we skip it for applying .editorconfig to files or something, but it's not blocked by the APIs to create a solution. Is there any chance one of those can get here?

Reply:
> Isn't it legal for an AnalyzerConfigDocument to have a non-absolute path?

It ends up throwing here:
https://github.com/dotnet/roslyn/blob/67549d9d6a30a44edefbbc38d8c0ba6db6228bb9/src/Compilers/Core/Portable/CommandLine/AnalyzerConfig.cs#L115

Reply:
Tagging @tmat who was also looking at this issue somewhere else too.

Reply:
I believe the compiler should not require full path. I don't see reason why relative eould not work. https://github.com/dotnet/roslyn/issues/41940